src/common/actions/register-name.js

```diff
@@ -198,3 +211,79 @@ export function registerNameClear(id) {
     }
   };
 }
+
+export async function internalNameOwnership(root, discharge, snapName) {
+  if (!snapName) {
+    throw new APICompatibleError({
```

Comment:
If this error is caused by developer mistake, is it really something we want to handle by keeping it in the store?